### ISSUE TITLE
Upload is broken in UploadController.php function upload

### ISSUE THREAD

**Issue by gletiecq** [linuxforphp/filebrowser#6]
**Describe the bug**
When a file is chosen to be uploaded system trows the following exception and the file fails to upload: 

 PHP Fatal error:  Uncaught Error: Call to a member function isValid() on array in /var/www/filebrowser/backend/Controllers/UploadController.php:71\nStack trace:\n#0 [internal function]: Filebrowser\\Controllers\\UploadController->upload()\n#1 /var/www/filebrowser/vendor/php-di/invoker/src/Invoker.php(82): call_user_func_array()\n#2 /var/www/filebrowser/vendor/php-di/php-di/src/Container.php(276): Invoker\\Invoker->call()\n#3 /var/www/filebrowser/backend/Services/Router/Router.php(81): DI\\Container->call()\n#4 /var/www/filebrowser/backend/App.php(34): Filebrowser\\Services\\Router\\Router->init()\n#5 /var/www/filebrowser/dist/index.php(95): Filebrowser\\App->__construct()\n#6 {main}\n  thrown in /var/www/filebrowser/backend/Controllers/UploadController.php on line 71, referer: http://192.168.0.110/

It would appear in the upload function that $filename is extracted from $request->input but that is not used later on to identify the file name used to build $file.  In playing with the system it appears that a forward slash is pre-pended on the file name and when isValid is called on that value, 

Screen displays the filename with a pre-pended forwad slash as in "/1920x1080p24_Contrast_Main-1024x576.png" above a red bar.

isValid() fails with a file does not exist error, but I can't be sure about this.  Too long since I've been playing with PHP.

**Environment (please complete the following information):**
 - FileBrowser Version - using filebrowser_latest pulled on 12/12/23
 - Server:  Apache 2.4.57 on Debian Linux (Linux avtech 6.1.0-rpi7-rpi-v8 #1 SMP PREEMPT Debian 1:6.1.63-1+rpt1 (2023-11-24) aarch64 GNU/Linux)
 - PHP Version PHP 8.2.7
 - Browser Chrome, also on Edge
 - Installed on a Raspberry Pi 4B (8GB) w/32GB SD Card to use as a portable SAN.

**Additional context**

**Comment by gletiecq** [comment]
In debgging the problem, editing the file upload limits in /etc/php/8.2/apache2 changed the behavior.  I no longer see an error indication, but the upload now fails silently.

**Comment by andrewscaya** [comment]
Hi @gletiecq,

FileBrowser is used on many accounts that have been running PHP 8.2 on the LfPHP Cloud Services for more than a year now, and we are not able to reproduce your issue. Also, we run all of our front end and back end tests against each major version of PHP. Therefore, the chances are pretty high that the issue is related to your setup.

Please make sure that you have all of the necessary extensions for PHP, that the 'repository' directory has the required 'write' permissions for the Web server, and that the JS front end is routing the request correctly to the controller (seems to be a routing issue from what I can see in the error messages mentioned in your post).

If have any further questions, please feel free to ask them.

Many thanks for your interest in the FileBrowser project!

-- Andrew

**Comment by Namtor-FFM** [comment]
Hi @andrewscaya,

I just installed Filebrowser on my local server and I exactly have the same issue. I can create a new folder and a new file, I can update the file, I can remove a file and a folder. However, when I want to upload a document, I get the same error message as mentioned before in the log file.

I'm using PHP 8.2.14 and I checked that all prerequisites are properly installed.
In configuration.php, I just did a copy of the sample.

In the previous answer, you said "[check that] the JS front end is routing the request correctly to the controller". What does it really mean?

Many thanks,
Nicolas
